go/vt/mysqlctl/binlogs_gtid.go

```diff
@@ -133,7 +132,16 @@ func ChooseBinlogsForIncrementalBackup(
 		}
 		return binaryLogsToBackup, incrementalBackupFromGTID, incrementalBackupToGTID, nil
 	}
-	return nil, "", "", vterrors.Errorf(vtrpc.Code_FAILED_PRECONDITION, "no binary logs to backup (increment is empty)")
+	if prevGTIDsUnion.Union(purgedGTIDSet).Equal(backupFromGTIDSet) {
+		// This means we've iterated over all binary logs, and as it turns out, the backup pos is
+		// identical to the Previous-GTIDs of the last binary log. But, we also know that we ourselves
+		// have flushed the binary logs so as to generate the new (now last) binary log.
+		// Which means, from the Pos of the backup till the time we issued FLUSH BINARY LOGS, there
+		// were no new GTID entries. The database was completely silent during that period.
```

Comment:
Slightly more explanative to note in this and similar places that the database performed no writes vs "did nothing" / "was silent" etc.

Reply:
Reworded.